src/Blocks/BlockPlant.h

```diff
@@ -0,0 +1,134 @@
+
+// BlockPlant.h
+
+// Base class for any growing block
+
+
+
+
+
+#pragma once
+
+#include "BlockHandler.h"
+
+class cBlockPlant : public cBlockHandler
+{
+	bool m_NeedLightToGrow;
+public:
+	cBlockPlant(BLOCKTYPE a_BlockType, bool a_LightToGrow)
+		: cBlockHandler(a_BlockType), m_NeedLightToGrow(a_LightToGrow){}
+
+protected:
+	/** These are returned by cBlockPlant::CanGrow() to determine what action needs to be
+	 taken by the block handler.
+	 */
+	enum PlantAction
+	{
+		paDeath,
+		paGrowth,
+		paStay
+	};
+
+	/** Checks whether there is enough light for the plant to grow.
+	 If the plant doesn't require light to grow, then it returns paGrowth.
+	 If the plant requires light to grow and there is enough light, it returns paGrowth.
+	 If the plant requires light to grow and there isn't enough light, it returns paStay.
+	 If the plant requires light to grow and there is too little light, it returns paDeath. */
```

Comment:
There are still extra spaces in the indent here. How come the check doesn't catch it?